### PR TITLE
Fix nested list counter

### PR DESCRIPTION
**Pull request by codeEmpress1**
## Done
- Fix nested list counter
[List of work items including drive-bys]

Fixes [list issues/bugs if needed]
https://github.com/canonical/vanilla-framework/issues/4998

## QA

- Open [demo](insert-demo-url)
- [Add QA steps]
- Review updated documentation:
  - [List any updated documentation for review]

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [ ] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [ ] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [ ] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [what's new page](https://github.com/canonical/vanilla-framework/blob/main/releases.yml).

### Before
image

### After
Screenshot from 2024-03-01 11-50-32

[if relevant, include a screenshot or screen capture]

### CODE REVIEW

File: scss/_patterns_lists.scss [hunk 1]
```diff
@@ -71,14 +71,16 @@ $list-step-bullet-margin: $sph--x-large;
   }
 
   %vf-list-item-ordered {
-    &::marker {
+      counter-reset: li;
```
Comment (bartaz) [inline scss/_patterns_lists.scss:74]: I think the changes you made here are in good direction, but are not necessary in this form.

Overall, I believe the bug comes from the fact that we use 2 different counters in different lists: we use `li` counter and `list-item` counter. The stepped list uses one of them (`li`) and the divided list uses the other (`list-item`).

I believe the difference is not necessary. We can use the same counter in both (so they correctly reset and increment). 

So I think the fix for the issue would be to change any references to counters to use the same one (either `li` or `list-item` - doesn't matter, just the same everywhere).
So any `counter-reset`, `counter-increment` or value of `counter(…)` should use the same (`list-item` for example).

File: templates/docs/examples/patterns/lists/list-nested-stepped-detailed.html [hunk 2]
```diff
@@ -0,0 +1,56 @@
+{% extends "_layouts/examples.html" %}
+{% block title %}Lists / Nested Stepped Detailed Ordered List{% endblock %}
```
Comment (bartaz) [inline templates/docs/examples/patterns/lists/list-nested-stepped-detailed.html:2]:
```suggestion
{% block title %}Lists / Ordered list nested in stepped list{% endblock %}
```

**Review by bartaz — APPROVED** [review]
LTGM, thanks a lot!

**Review by bartaz — CHANGES_REQUESTED** [review]
There is some issue with standard stepped list now, that it increases by 2 on each level:
https://vanilla-framework-5011.demos.haus/docs/patterns/lists#vertical-stepped

image

I haven't checked in detail yet, but maybe `counter-increment` happens twice or something like that.

**Comment by codeEmpress1** [comment]
> There is some issue with standard stepped list now, that it increases by 2 on each level: https://vanilla-framework-5011.demos.haus/docs/patterns/lists#vertical-stepped
> 
> image
> I haven't checked in detail yet, but maybe `counter-increment` happens twice or something like that.

Ohh, Yea. It happens twice for this, I'll look at it

**Comment by bartaz** [comment]
@codeEmpress1 It seems the issue was that we've been using default browser `list-item` counter, which is also increased natively by the browser. So the counter was increased both by `li` element, and by `__title` class.

I moved counter increase to `li` element, so this only happens there and not in the title as well. This seems to help.

Please review/QA if all looks good from your side. Thanks.

**Comment by codeEmpress1** [comment]
LGTM!

**Pull request merged**